plugins/modules/vmware_object_rename.py

```diff
@@ -46,6 +46,34 @@
       required: True
       aliases: ['object_new_name']
       type: str
+    filter:
+      description:
+      - Filter an object to search.
+      - This module can't get over 1,000 objects at onece, so you should use the I(filter) parameter
+        if VCSA has over 1,000 objects with the specified object type.
+      type: dict
+      default:
+        datacenters: []
+        folders: []
+        names: []
+      version_added: '1.14.0'
+      suboptions:
+        datacenters:
+          description:
+          - Specify moid of the datacenter where objects exist.
+          type: list
+          elements: str
+        folders:
```

Comment:
Do you think we should add other filters as well? Since we are here.

Reply:
Thanks, @Akasurde for looking at this PR, and reviewing it!  
This table is a summary that the parameters supported by the filter of each object.

![image](https://user-images.githubusercontent.com/19516126/130454505-08c590a4-7e88-4fce-b233-04b6a1ac6a9b.png)

The datacenters, names, and folders parameters have a common in all objects. (ResourcePool is an exception when the folders parameter, but...)  
clusters, type, and hosts also have a common in some objects, but I have a concern to be complex if implementing the parameters in the module. (also include the rest of individual parameters)  

I think it's good to keep it simple for now and add those parameters when the demand for them arises.  
What do you think?

Reply:
Hi @Akasurde 
I could confirm that I can filter an object to be less than 1,000 objects with the filter spec for a standard SDK even not adding the filter parameter.
So I deleted the filter parameter I added.